android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/ExtensionsProcessDisabledPromptObserver.kt

```diff
@@ -23,23 +23,25 @@ open class ExtensionsProcessDisabledPromptObserver(
     private val store: BrowserStore,
     private val onShowExtensionsProcessDisabledPrompt: () -> Unit = { },
 ) : LifecycleAwareFeature {
-    private var promptScope: CoroutineScope? = null
+    private var scope: CoroutineScope? = null
 
     override fun start() {
-        promptScope = store.flowScoped { flow ->
-            flow.distinctUntilChangedBy { it.showExtensionsProcessDisabledPrompt }
-                .collect { state ->
-                    if (state.showExtensionsProcessDisabledPrompt) {
-                        // There should only be one active dialog to the user when the extensions
-                        // process spawning is disabled.
-                        onShowExtensionsProcessDisabledPrompt()
+        if (scope == null) {
+            scope = store.flowScoped { flow ->
+                flow.distinctUntilChangedBy { it.showExtensionsProcessDisabledPrompt }
+                    .collect { state ->
+                        if (state.showExtensionsProcessDisabledPrompt) {
+                            onShowExtensionsProcessDisabledPrompt()
+                        }
                     }
-                }
+            }
         }
     }
 
+    /**
+     * This observer needs to run indefinitely.
```

Comment:
```suggestion
     * This observer needs to run indefinitely because we need to react to state changes
     * when the app is in the background too.
```